monai/networks/nets/milmodel.py

```diff
@@ -0,0 +1,184 @@
+import os
+
+import torch
+import torch.nn as nn
+import torchvision.models as models
+
+
+class MILModel(nn.Module):
+    """
+    A wrapper around backbone classification model suitable for MIL
+
+    Args:
+        num_classes: number of output classes
+        mil_mode: MIL variant (supported max, mean, att, att_trans, att_trans_pyramid
+        pretrained: init backbone with pretrained weights. Defaults to True.
+        backbone: Backbone classifier CNN. Defaults to None, it which case ResNet50 will be used.
+        backbone_nfeatures: Number of output featues of the backbone CNN (necessary only when using custom backbone)
+
+    mil_mode:
+        mean - average features from all instances, equivalent to pure CNN (non MIL)
+        max - retain only the instance with the max probability for loss calculation
+        att - attention based MIL https://arxiv.org/abs/1802.04712
+        att_trans - transformer MIL
+        att_trans_pyramid - transformer pyramid MIL
+
+    """
+
+    def __init__(self, num_classes, mil_mode="att", pretrained=True, backbone=None, backbone_nfeatures=None):
+
+        super().__init__()
+
+        self.mil_mode = mil_mode
+        print("MILModel with mode", mil_mode, "num_classes", num_classes)
+        n_trans = 4
```

Comment:
Does this number depends on the model being used?
If yes, shouldn't be derived from the model?
If no, can it be provided as an input argument?

Reply:
it doesn't depend on the model.  yes we can make it an argument

Reply:
I made it an argument